internal/action/main_test.go

```diff
@@ -0,0 +1,291 @@
+	}
+
+	assert.Equal(t, ws.TerraformVersion, "1.0.4")
+
+	v, err := client.Variables.Create(ctx, ws.ID, tfe.VariableCreateOptions{
+		Key:      strPtr("foo"),
+		Value:    strPtr("bar"),
+		Category: tfe.Category(tfe.CategoryTerraform),
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, v.Value, "bar")
+
+	Run(config)
+
+	ws, err = client.Workspaces.Read(ctx, config.Organization, config.Name)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, ws.TerraformVersion, "1.0.5")
+
+	v, err = client.Variables.Read(ctx, ws.ID, v.ID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, v.Value, "baz")
+
+	t.Cleanup(func() {
+		if err := RemoveTestWorkspaces(ctx, client, config.Organization, workspacePrefix); err != nil {
+			t.Fatal(err)
+		}
+	})
+}
+
+func TestDriftCorrection(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	ctx := context.Background()
+
+	config := NewTestRunConfig()
+
+	client, err := tfe.NewClient(&tfe.Config{
+		Address: fmt.Sprintf("https://%s", config.Host),
+		Token:   config.Token,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := RemoveTestWorkspaces(ctx, client, config.Organization, workspacePrefix); err != nil {
+		t.Fatal(err)
+	}
+
+	ws, err := client.Workspaces.Create(ctx, config.Organization, tfe.WorkspaceCreateOptions{
+		Name:             &config.Name,
+		TerraformVersion: &config.TerraformVersion,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	v, err := client.Variables.Create(ctx, ws.ID, tfe.VariableCreateOptions{
+		Key:      strPtr("foo"),
+		Value:    strPtr("bar"),
+		Category: tfe.Category(tfe.CategoryTerraform),
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	Run(config)
+
+	_, err = client.Variables.Read(ctx, ws.ID, v.ID)
+	if err == nil {
+		t.Fatal("Expected variable not to exist")
+	}
+
+	if err.Error() != "resource not found" {
+		t.Fatalf("Expected error to be resource not found: %s", err)
+	}
+
+	t.Cleanup(func() {
+		if err := RemoveTestWorkspaces(ctx, client, config.Organization, workspacePrefix); err != nil {
+			t.Fatal(err)
+		}
+	})
+}
+
+func TestMultipleWorkspaces(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	ctx := context.Background()
+
+	config := NewTestRunConfig()
+
+	config.Workspaces = `---
+- staging
+- production`
+
+	config.WorkspaceVariables = `---
+staging:
+  - key: environment
+    value: staging
+    category: env
+production:
+  - key: environment
+    value: production
+    category: env`
+
+	client, err := tfe.NewClient(&tfe.Config{
+		Address: fmt.Sprintf("https://%s", config.Host),
+		Token:   config.Token,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := RemoveTestWorkspaces(ctx, client, config.Organization, workspacePrefix); err != nil {
+		t.Fatal(err)
+	}
+
+	ws, err := client.Workspaces.List(ctx, config.Organization, tfe.WorkspaceListOptions{
+		Search: &workspacePrefix,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, len(ws.Items), 0)
```

Comment:
Backwards, `Equal` args are `expected, actual`

```suggestion
	assert.Len(t, ws.Items, 0)
```

Of course `Len` is reversed...

More specific assertion methods generally get you better error messages without having to format them yourself. In this case, you'll get something closer to "expected length of 0, got 1" instead of "expected 0 to equal 1."